driver/src/tracking_references.cpp

```diff
@@ -0,0 +1,491 @@
+        //     };
+
+        //     m_pSocketComm->Send(
+        //         &resp,
+        //         sizeof(resp)
+        //     );
+
+        //     DriverLog("tracking reference: queue for buffer size processed");
+        //     break;
+        // }
+
+        case EManagerMsgType_ipd: {
+            vr::VRSettings()->SetFloat(
+                k_pch_Hmd_Section,
+                k_pch_Hmd_IPD_Float,
+                message->data.ipd.ipd_meters
+            );
+
+            HoboVR_ManagerResp_t resp{EManagerResp_ok};
+            m_pSocketComm->Send(
+                &resp,
+                sizeof(resp)
+            );
+            DriverLog("tracking reference: ipd change request processed");
+            break;
+        }
+
+        case EManagerMsgType_uduString: {
+            std::string temp;
+            for (int i=0; i < message->data.udu.len; i++) {
+                if (message->data.udu.devices[i] == 0)
+                    temp += "h";
+                else if (message->data.udu.devices[i] == 1)
+                    temp += "c";
+                else if (message->data.udu.devices[i] == 2)
+                    temp += "t";
+                else if (message->data.udu.devices[i] == 3)
+                    temp += "g";
+            }
+            _vpUduChangeBuffer = temp;
+
+            // vr::VREvent_Notification_t event_data = {20, 0};
+            // vr::VRServerDriverHost()->VendorSpecificEvent(
+            //     m_unObjectId,
+            //     vr::VREvent_VendorSpecific_HoboVR_UduChange,
+            //     (vr::VREvent_Data_t&)event_data,
+            //     0
+            // );
+            _UduEvent = true;
+
+            DriverLog(
+                "tracking reference: udu settings change request processed"
+            );
+            HoboVR_ManagerResp_t resp{EManagerResp_ok};
+            m_pSocketComm->Send(
+                &resp,
+                sizeof(resp)
+            );
+            break;
+        }
+
+        case EManagerMsgType_setSelfPose: {
+
+            m_Pose.vecPosition[0] = message->data.self_pose.position[0];
+            m_Pose.vecPosition[1] = message->data.self_pose.position[1];
+            m_Pose.vecPosition[2] = message->data.self_pose.position[2];
+
+            if (m_unObjectId != vr::k_unTrackedDeviceIndexInvalid) {
+                vr::VRServerDriverHost()->TrackedDevicePoseUpdated(
+                    m_unObjectId,
+                    m_Pose,
+                    sizeof(m_Pose)
+                );
+            }
+
+            HoboVR_ManagerResp_t resp{EManagerResp_ok};
+            m_pSocketComm->Send(
+                &resp,
+                sizeof(resp)
+            );
+            DriverLog("tracking reference: pose change request processed");
+            break;
+        }
+
+        case EManagerMsgType_distortion: {
+            float newK1 = (float)message->data.distortion.k1;
+            float newK2 = (float)message->data.distortion.k2;
```

Comment:
Looking at the distortion parameters for steam, I think we might need the directx memory mapping before we can use any special lenses like holographic. (Or vulkan, but that might just be directx before going through an emulator on linux.)
Once we have that, we'd send that through a gpu operation (glsl, vulkan, or directx), to apply whatever custom lens distort function we want.

I think we should have a dependency map, link the other repo that has directx+wifi for wireless as a dependency, then test that on linux, then attempt different lens projections.